src/facade/dragonfly_connection.cc

```diff
@@ -313,6 +313,10 @@ Connection::Connection(Protocol protocol, util::HttpListenerBase* http_listener,
 
   migration_enabled_ = absl::GetFlag(FLAGS_migrate_connections);
 
+  // Create shared_ptr with empty value and associate it with `this` pointer (aliasing constructor).
+  // We use it for reference counting and accessing `this` (without managing it).
+  self_ = {std::make_shared<std::monostate>(std::monostate{}), this};
```

Comment:
nice trick :)

Reply:
I still need @romange 's opinion if he's okay with it instead of writing a custom primitive 🙂

Reply:
fwiw I don't see a reason why it shouldn't work :upside_down_face: